hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -405,7 +430,11 @@ void cleanup(NodeEngine nodeEngine) {
                       .collect(Collectors.toSet())
                       .forEach(id -> {
                           jobMetrics.delete(id);
-                          jobResults.delete(id);
+                          JobResult jobResult = jobResults.get(id);
+                          if (jobResult != null) {
+                              jobResult.destroy(instance);
```

Comment:
so observables are only cleared if there's more than 1000 of them? That doesn't sound good enough to me. There's also TTL of 7 days on the JobResults, but if it's cleared this way then the Ringbuffer will actually will not be deleted.

Reply:
Yes,  crap... I again accepted indications without really thinking about them... Have reverted the code to a past version which does `Observable` cleanup separately from `JobResults` clean-up. Will do further changes to it for other TODOs.